File: ms-contas/src/main/java/com/pocauditoria/contas/application/dto/AuditEventDTO.java
package com.pocauditoria.contas.application.dto;

import java.time.Instant;
import java.util.List;
import java.util.Map;
import lombok.Builder;

@Builder
public record AuditEventDTO(
        String id,
        Instant timestamp,
        String operation,
        String entityName,
        String entityId,
        String userId,
        Map<String, Object> oldValues,
        Map<String, Object> newValues,
        List<String> changedFields,
        String sourceService,
        String correlationId
) {
}

File: ms-transacoes/src/MsTransacoes.Application/DTOs/AuditEventDTO.cs
namespace MsTransacoes.Application.DTOs;

public sealed class AuditEventDTO
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public DateTime Timestamp { get; set; } = DateTime.UtcNow;
    public string Operation { get; set; } = string.Empty;
    public string EntityName { get; set; } = string.Empty;
    public string EntityId { get; set; } = string.Empty;
    public string UserId { get; set; } = string.Empty;
    public Dictionary<string, object?>? OldValues { get; set; }
    public Dictionary<string, object?>? NewValues { get; set; }
    public List<string>? ChangedFields { get; set; }
    public string SourceService { get; set; } = "ms-transacoes";
    public string? CorrelationId { get; set; }
}
